### PR TITLE
[RISCV][MISched] Set EnableIntervals to true for SiFive7

### PR DESCRIPTION
**Pull request by michaelmaitland**
The SiFive7 scheduler model has been using AcquireAtCycles and ReleaseAtCycles for some time. Without EnableIntervals, the scheduler was not making decisions based on this information. This patch sets EnableIntervals to true, and the test case demonstrates that the VADD instructions can be issued one cycle earlier since the VCQ is not reserved. This leads to better saturation of the VA.

I wonder whether it would be better (in a future patch) to move EnableIntervals out of the scheduler model and add an option like `use-intervals-to-schedule` so that we can control per-program basis. EnableIntervals isn't really a property of the scheduler model itself, and more a property of how the scheduler behaves.

FWIW, I had no idea EnableIntervals existed, and when I added AcquireAtCycles and ReleaseAtCycles, I totally assumed that the scheduler had been using this information out of the box.

### CODE REVIEW

**Review by fpetrogalli — APPROVED** [review]

**Comment by fpetrogalli** [comment]
> I wonder whether it would be better (in a future patch) to move EnableIntervals out of the scheduler model and add an option like use-intervals-to-schedule so that we can control per-program basis. EnableIntervals isn't really a property of the scheduler model itself, and more a property of how the scheduler behaves.

Please create an [issue](https://github.com/llvm/llvm-project/issues), we will discuss it there.

Thank you!

File: llvm/test/CodeGen/RISCV/sifive7-enable-intervals.mir [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# RUN: llc -mtriple=riscv64 -mcpu=sifive-x280 -run-pass=machine-scheduler      \
+# RUN:  -debug-only=machine-scheduler -misched-dump-schedule-trace             \
+# RUN:  -misched-topdown -o - %s 2>&1 | FileCheck %s
+
+# The purpose of this test is to show that the VADD instructions are issued at
```
Comment (wangpc-pp) [inline llvm/test/CodeGen/RISCV/sifive7-enable-intervals.mir:5]: `are issued at` where? Or this is a special English phrase?

**Review by wangpc-pp — APPROVED** [review]
LGTM.

File: llvm/test/CodeGen/RISCV/sifive7-enable-intervals.mir (hunk 1 above)
Comment (michaelmaitland) [reply llvm/test/CodeGen/RISCV/sifive7-enable-intervals.mir:5]: `i` in the table below signifies the cycle that a VADD instruction `is issued at`. If we want to talk about multiple VADD instructions, then `i`s in the table below signify the cycles that the `VADD` instructions `are issued at`.

Comment (michaelmaitland) [reply llvm/test/CodeGen/RISCV/sifive7-enable-intervals.mir:5]: I will drop the word `at` so it makes sense.

**Comment by chapuni** [comment]
Does it fail in -Asserts?

**Comment by michaelmaitland** [comment]
> Does it fail in -Asserts?

I haven’t seen any failures. Are you asking for any specific reason?

**Comment by chapuni** [comment]
https://lab.llvm.org/buildbot/#/builders/67/builds/13712

**Comment by zhuhan0** [comment]
> https://lab.llvm.org/buildbot/#/builders/67/builds/13712

We're seeing the same failure as shown in this link.

**Comment by petrhosek** [comment]
We're seeing the same test failure on our builders, can we revert this change?

**Comment by dyung** [comment]
We are also seeing this failure on our internal release build with assertions disabled.

**Comment by ornata** [comment]
I think this needs `REQUIRES: asserts`

**Comment by ornata** [comment]
For now, I think we should revert. What's the process for that now that we're on PRs? PR for the revert?

**Comment by michaelmaitland** [comment]
@nikic thanks for the fix. Sorry to all who had broken builds. I did not see your messages until I woke up.

**Pull request merged**